### PR TITLE
Add edge cases tests for regex langlib functions

### LINKED ISSUE #39748: [Improvement]: Add test cases with edge cases in regex langlibs
### Description

${subject}

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes #39748

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by SasinduDilshara** [comment]
Marked this as a draft because several regex related issues ned to fix before merging this.

File: langlib/langlib-test/src/test/resources/test-src/regexp_test.bal [hunk 1]
```diff
@@ -61,6 +61,61 @@ function testFind() {
...
+    
+    var regExpr7 = re `${"3*"}`;
+    regexp:Span? res71  = regExpr7.find("1233333333", 5);
+    assertTrue(res71 is regexp:Span);
+    regexp:Span resultSpan81 = <regexp:Span>res71;
+    assertEquality(5, resultSpan81.startIndex);
+    assertEquality(10, resultSpan81.endIndex);
+    assertEquality("33333", resultSpan81.substring());
+
+   //  regexp:Span? res72  = regExpr7.find("");
```
Comment (pcnfernando) [inline langlib/langlib-test/src/test/resources/test-src/regexp_test.bal:81]: tag the relevant PR in the commented sections

Comment (SasinduDilshara) [reply langlib/langlib-test/src/test/resources/test-src/regexp_test.bal:81]: Will do

**Review by LakshanWeerasinghe — APPROVED** [review]
LGTM!

**Comment by LakshanWeerasinghe** [comment]
We have identified several issues related to langlib functions with different patterns like,
1.  CharSetRanges with escaped ranges.
2. Texts with unicodes.

All of the cases should be tested with patterns created at runtime and compile time.

**Review by LakshanWeerasinghe — CHANGES_REQUESTED** [review]
Need to address the comments.
